### PR TITLE
 Add popovers for citations in the HTML documentation

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This PR adds functionality to the Doxygen HTML documentation so that when hovering over a citation a popover with bibliographic information appears. It contains information such as authors, title and URL/eprint link:

bildschirmfoto 2018-12-13 um 02 17 46

### Types of changes:

- [x] New feature

### Component:

- [x] Documentation

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

Since the code in this PR is not our usual C++, I am unsure on the formatting guidelines. In particular our git hooks complain about long lines in a variety of files, as well as our clang-format configuration not support JavaScript. I applied some judgement, but please let me know yours.

### CODE REVIEW

**Review by wthrowe — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Is this error anything to worry about?
```
jquery.js:4 Access to XMLHttpRequest at 'file:///spectre/build_doc/pr_1242/citelist.html' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.
```
I get this when I render locally, but if this won't happen on the public doc I'm not sure I care. I don't know anything about JS or web development so I can't make any useful comment.

**Comment by hrueter** [comment]
@nilsdeppe Your browser thinks, that a javascript tries to read your local files and blocks this access.
The solution would be to run a webserver locally. See e.g. here: https://github.com/mrdoob/three.js/wiki/How-to-run-things-locally

If possible we should just ignore it locally. Any online doc should work fine.

**Comment by nilsdeppe** [comment]
@hrueter thanks for the explanation! I'm fine with ignoring this locally :)

@kidder this is ready to go

**Comment by nilsvu** [comment]
@hrueter @nilsdeppe Exactly, this is what the comment in the `spectre.js` is about. Just run a `python3 -m http.server` in the `html` directory to try locally.

**Comment by nilsdeppe** [comment]
Oooh fancy! I can confirm it works locally just as it is documented :D Nice job @nilsleiffischer!

File: docs/config/js/spectre.js [hunk 1]
```diff
@@ -63,4 +63,49 @@ window.onload = function(){
...
+    $("body").children().find('a[href*="citelist.html#CITEREF_"]').each(function () {
+        var tooltip_base = $(this);
+        // Get the reference id, e.g. `CITEREF_Kopriva`. This is the one used as
+        // anchor on the bibliography page.
+        var ref_id = $(this).attr('href').match(/CITEREF_([a-zA-Z0-9]+)/)[0];
+        // Load the bibliography page to retrieve the reference data as nicely
+        // formatted HTML.
+        // This does not work locally because it is forbidden to access files,
+        // so just spin up a web server in the `html` directory with
+        // `python3 -m http.server` to enable this functionality
```
Comment (kidder) [inline docs/config/js/spectre.js:77]: can you add something about this command also in docs/DevGuide/Travis.md in the subsection on `How to perform the checks locally` in the `Documentation` item.

Comment (nilsvu) [reply docs/config/js/spectre.js:77]: done

**Comment by kidder** [comment]
I am happy for you to squash

**Comment by nilsvu** [comment]
done @sxs-collaboration/spectre-core-devs

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]
There are issues the the popovers appearing off-screen sometimes, but that's probably a harder problem to solve and I think this is fine for a first pass.

**Comment by nilsvu** [comment]
@wthrowe could you open an issue with a screenshot when you get the chance?

**Review by kidder — APPROVED** [review]

**Pull request merged**